File: Client/WebClient/src/app/models/networkAlarm.ts
import { NetworkEventDto } from "./dtos/networkEventDto";

export class NetworkAlarm {
  rtuId: string;
  channel: string; // Main or Reserve
  hasBeenSeen: boolean;
}

export class NetworkAlarmIndicator {
  list: NetworkAlarm[] = [];

  public NetworkEventReceived(signal: NetworkEventDto): string {
    const alarm = this.list.find((a) => a.rtuId === signal.rtuId);
    return "ok";
  }
}

File: Shared/Dto/ForWebApi/NetworkAlarm.cs
using System;
using System.Collections.Generic;

namespace Iit.Fibertest.Dto
{
    public class NetworkAlarm
    {
        public int EventId;
        public Guid RtuId;
        public string Channel; // Main or Reserve
        public bool hasBeenSeen;
    }

    public class OpticalAlarm
    {
        public int SorFileId;
        public Guid TraceId;
        public bool hasBeenSeen;
    }

    public class BopAlarm
    {
        public int EventId;
        public string Serial;
        public bool hasBeenSeen;
    }

    public class AlarmsDto
    {
        public List<NetworkAlarm> NetworkAlarms;
        public List<OpticalAlarm> OpticalAlarms;
        public List<BopAlarm> BopAlarms;
    }
}
